magic-gds-drc/Dockerfile

```diff
@@ -0,0 +1,49 @@
+# Copyright 2021 Efabless Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# SPDX-License-Identifier: Apache-2.0
+FROM ubuntu:20.04
+
+ENV DEBIAN_FRONTEND noninteractive
+
+RUN apt-get update
+RUN apt-get install -y build-essential git curl python3 python3-pip
+
+# Setup Magic
+ENV MAGIC_TAG 8.3.160
+RUN apt-get install -y tcl-dev tk-dev csh
+WORKDIR /tmp/magic
+RUN curl -L https://github.com/RTimothyEdwards/magic/archive/refs/tags/${MAGIC_TAG}.tar.gz | tar -xzC /tmp/magic --strip-components=1
```

Comment:
Try grouping commands in a single `RUN` statement, for avoiding unnecessary layers in the image.

Reply:
Honest question: how harmful are the unnecessary layers? For me, more layers is a good thing because that means more granular caching. I understand that's not really relevant for CI but it greatly eases the development process locally.

Reply:
There is a hard limit of 127 layers: https://github.com/docker/docker.github.io/issues/8230#issuecomment-468630187. Since you never know the exact number of layers of the base image, or how many layers will consumers of your images add, it is good practice to be conservative and minimize the number of layers you add to the stack. 

It is ok to have multiple RUN statements. Don't need to be radical and user one only. However, "configure, make, make install" is conceptually an atomic operation. There is little benefit in splitting those into layers. 

Find this and similar issues/concepts in https://docs.docker.com/develop/develop-images/dockerfile_best-practices/.

Reply:
Ah, gotcha. Thanks for the info.

Reply:
Done